### PR TITLE
Refactoring StatisticalAnalyser class

### PR DESCRIPTION
**Pull request by rmilfont**
Description:Change methods to return parameters separated instead of an array of parameters
Commit:Ref

### CODE REVIEW

File: src/StatisticalAnalyser.java [hunk 1]
```diff
@@ -15,55 +15,50 @@ public StatisticalAnalyser(Graph graph, Map<Vertex, ArrayList<Region>> regionsFo
...
-        average = sum / regSizes.size();
-
-        stats[0] = (double) regSizes.get(regSizes.size() - 1);
-        stats[1] = (double) regSizes.get(0);
-        stats[2] = (double) average;
-        return stats;
+        return regSizes;
     }
 
     public double linkWeightMean(ArrayList<ArrayList<Path>> paths) {
```
Comment (jmferreiratech) [inline src/StatisticalAnalyser.java:58]: Be more consistent about using mean or average. Both in the method and variable names.

File: src/StatisticalAnalyser.java [hunk 2]
```diff
@@ -82,57 +77,63 @@ public double pathWeightMean(ArrayList<ArrayList<Path>> paths) {
         return acc*linkWeightMean(paths)/(double)paths.size();
     }
 
-    public double[] pathWeightStats(ArrayList<ArrayList<Path>> paths) {
-        double[] stats = new double[2];
-        double acc = 0;
+    public double pathWeightStdDeviation(ArrayList<ArrayList<Path>> paths) {
+        int nPaths = 0;
+        double acc = 0.0; double dev;
```
Comment (jmferreiratech) [inline src/StatisticalAnalyser.java]: Avoid putting more than one statement per line.

File: src/StatisticalAnalyser.java [hunk 3]
```diff
@@ -15,55 +15,50 @@ public StatisticalAnalyser(Graph graph, Map<Vertex, ArrayList<Region>> regionsFo
...
+                    * (link.weight() - mean);
 
-        mean = linksWeight / (double) graph.getEdges().size();
-        stats[0] = mean;
+        int nEdges = graph.getEdges().size();
+        return  Math.sqrt(temp / (double) nEdges);
+    }
 
-        double temp = 0.0;
+    public double linkWeightAverage() {
```
Comment (jmferreiratech) [inline src/StatisticalAnalyser.java]: Don't you think averageLinkWeight sound better?

File: src/StatisticalAnalyser.java [hunk 4]
```diff
@@ -15,55 +15,50 @@ public StatisticalAnalyser(Graph graph, Map<Vertex, ArrayList<Region>> regionsFo
...
-    // Link weight stats [0] - mean / [1] - standard deviation
-    public double[] linkWeightStats() {
-        double linksWeight = 0.0;
-        double[] stats = new double[2];
-        double mean = 0.0;
-        double std = 0.0;
+
+    public double linkWeightStdDeviation() {
+        double temp = 0.0;
+        double mean = linkWeightAverage();
```
Comment (jmferreiratech) [inline src/StatisticalAnalyser.java]: Get ride of this blank line.

File: src/StatisticalAnalyser.java [hunk 5]
```diff
@@ -15,55 +15,50 @@ public StatisticalAnalyser(Graph graph, Map<Vertex, ArrayList<Region>> regionsFo
...
+    public double averageNumberOfRegions() {
+        int sum = 0;
+        for (int size : regSizes())
+            sum += size;
+        return (double)(sum / regSizes().size());
+    }
 
-        double[] stats = new double[3];
-        double average;
+    private List<Integer> regSizes() {
```
Comment (jmferreiratech) [inline src/StatisticalAnalyser.java]: You could find a better name to this method. Something like numberOfRegionsPerVertex or even regionsPerVertex...

File: src/StatisticalAnalyser.java [hunk 6]
```diff
@@ -82,57 +77,63 @@ public double pathWeightMean(ArrayList<ArrayList<Path>> paths) {
...
-        double[] stats = new double[2];
-        double acc = 0;
+    public double pathWeightStdDeviation(ArrayList<ArrayList<Path>> paths) {
+        int nPaths = 0;
+        double acc = 0.0; double dev;
+        double average = pathWeightAverage(paths);
+        for(ArrayList<Path> alp : paths) {
+            nPaths += alp.size();
+            for(Path path: alp) {
+                dev = path.getWeight() - average;
```
Comment (jmferreiratech) [inline src/StatisticalAnalyser.java]: dev could have a more local scope.

File: src/StatisticalAnalyser.java [hunk 7]
```diff
@@ -15,55 +15,50 @@ public StatisticalAnalyser(Graph graph, Map<Vertex, ArrayList<Region>> regionsFo
...
     }
-    // Link weight stats [0] - mean / [1] - standard deviation
-    public double[] linkWeightStats() {
-        double linksWeight = 0.0;
-        double[] stats = new double[2];
-        double mean = 0.0;
-        double std = 0.0;
+
+    public double linkWeightStdDeviation() {
+        double temp = 0.0;
```
Comment (jmferreiratech) [inline src/StatisticalAnalyser.java]: Try to use more meaningful variable names than 'temp', 'sum' and 'acc'.

**Review by jmferreiratech — CHANGES_REQUESTED** [review]
Please, could you also squash the imports with *? 
You can make a new refactoring commit with the changes requested...

File: src/StatisticalAnalyser.java (hunk 7 above)
Comment (rmilfont) [reply src/StatisticalAnalyser.java]: Done

File: src/StatisticalAnalyser.java (hunk 3 above)
Comment (rmilfont) [reply src/StatisticalAnalyser.java]: Done for this and some others.

File: src/StatisticalAnalyser.java (hunk 5 above)
Comment (rmilfont) [reply src/StatisticalAnalyser.java]: Agree.

File: src/StatisticalAnalyser.java (hunk 1 above)
Comment (rmilfont) [reply src/StatisticalAnalyser.java:58]: I didn't modify linkWeightMean and pathWeightMean methods that receives an A<A<Paths>>.

File: src/StatisticalAnalyser.java (hunk 2 above)
Comment (rmilfont) [reply src/StatisticalAnalyser.java]: Done.

File: src/StatisticalAnalyser.java (hunk 6 above)
Comment (rmilfont) [reply src/StatisticalAnalyser.java]: Make sense, in fact, in some cases, the remotion of "dev" is better.

**Pull request merged**